### PR TITLE
Use new macOS file-system event delivery in file-system watching

### PR DESCRIPTION
**Pull request by lptr**
This bumps `native-platform` to include:

- https://github.com/gradle/native-platform/pull/335

  The [FSEventStreamScheduleWithRunLoop](https://developer.apple.com/documentation/coreservices/1447824-fseventstreamschedulewithrunloop?language=objc) API we've been using has been deprecated since macOS 13. We need to switch to [FSEventStreamSetDispatchQueue](https://developer.apple.com/documentation/coreservices/1444164-fseventstreamsetdispatchqueue?language=objc).

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle test this please.

File: subprojects/distributions-dependencies/build.gradle.kts [hunk 1]
```diff
@@ -27,7 +27,7 @@ val jaxbVersion = "3.0.0"
 val junit5Version = "5.8.2"
 val mavenVersion = "3.9.5"
 val mavenResolverVersion = "1.9.16" // Should remain in-sync with `mavenVersion`
-val nativePlatformVersion = "0.22-milestone-25"
+val nativePlatformVersion = "0.22-snapshot-20240226084139+0000"
```
Comment (ljacomet) [inline subprojects/distributions-dependencies/build.gradle.kts:30]: Can we use anything but a snapshot?

Comment (lptr) [reply subprojects/distributions-dependencies/build.gradle.kts:30]: Sorry, didn't realize this would trigger you. Converted back to a draft.

**Comment by lptr** [comment]
@bot-gradle help

**Comment by lptr** [comment]
@bot-gradle run `AFT`.

**Comment by lptr** [comment]
@bot-gradle test AllFunctionalTestsPullRequestFeedback.

**Review by asodja — APPROVED** [review]
LGTM

**Comment by lptr** [comment]
@bot-gradle merge this please.

**Pull request merged**